### DISCUSSION TITLE
Can Foxglove WebSocket be supported?

### DISCUSSION THREAD

**Discussion by tokamakz** [eProsima/DDS-Record-Replay#174] [Q&A] [answered]
Using ddsrecorder can only save topic data as an MCAP file for offline visualization. ddsrecorder can add a WebSocket Server to convert topic data into MCAP data in real-time, thereby achieving online visualization.

reference:
https://docs.foxglove.dev/docs/connecting-to-data/frameworks/custom#foxglove-websocket
https://github.com/foxglove/ws-protocol/tree/main/cpp

**Comment by juanlofer-eprosima** [comment, marked as answer]
Hi @tokamakz ,

Thanks for reaching out. Yes we are aware of this capability in Foxglove, but unfortunately we are currently not able to allocate the resources required to implement the feature you are requesting. I suggest however contacting our [commercial support](mailto:info@eprosima.com) team in order to study your use case and offer a working solution.

**Reply by tokamakz** [reply]
I can help you develop this feature
